java/openjdk8-bootstrap/Portfile

```diff
@@ -5,49 +5,68 @@ PortSystem          1.0
-                    size    257863467
-
-depends_extract     port:xar
-
-extract {
-    system -W ${workpath} "mkdir -p ${worksrcpath}/packages"
-    file copy ${distpath}/${distfiles} ${worksrcpath}/packages
-    system -W ${workpath} "mkdir -p ${worksrcpath}/dmg"
-    system -W ${workpath} "mkdir -p ${worksrcpath}/pkg"
-    system -W ${workpath} "mkdir -p ${worksrcpath}/files"
-    system -W ${worksrcpath}/packages "hdiutil attach ${worksrcpath}/packages/${distfiles} -mountroot ${worksrcpath}/dmg"
-    system -W ${worksrcpath}/dmg "cp ${worksrcpath}/dmg/*/*.pkg ${worksrcpath}/pkg"
-    system -W ${worksrcpath}/dmg "hdiutil detach ${worksrcpath}/dmg/*"
-    system -W ${worksrcpath}/pkg "mv ${worksrcpath}/pkg/*.pkg ${worksrcpath}/pkg/jdk${version}.pkg"
-    system -W ${worksrcpath}/pkg "${prefix}/bin/xar -xf ${worksrcpath}/pkg/jdk${version}.pkg -C ${worksrcpath}/files"
-    system -W ${worksrcpath} "cpio -i < ${worksrcpath}/files/jdk180201.pkg/Payload"
+homepage            https://openjdk.org/
+
+if {${configure.build_arch} eq "x86_64"} {
+    set ver 1.8.0_201
+    master_sites        ${homepage}webapps/download/${distfiles}?BundleId=236866_42970487e3af4f5aa5bca3f542482c60&dummy=
+    distfiles           AutoDL
+    
+    checksums           rmd160  b237dd06774127454df314a28fa6b297b50a0dbf \
+                        sha256  5b52df4b3efd51480c40ad5abfa18a075365afbb2040bbc5e9f2db568435a6e2 \
+                        size    257863467
+
+    depends_extract     port:xar
+
+    extract {
+        system -W ${workpath} "mkdir -p ${worksrcpath}/packages"
+        file copy ${distpath}/${distfiles} ${worksrcpath}/packages
+        system -W ${workpath} "mkdir -p ${worksrcpath}/dmg"
+        system -W ${workpath} "mkdir -p ${worksrcpath}/pkg"
+        system -W ${workpath} "mkdir -p ${worksrcpath}/files"
+        system -W ${worksrcpath}/packages "hdiutil attach ${worksrcpath}/packages/${distfiles} -mountroot ${worksrcpath}/dmg"
+        system -W ${worksrcpath}/dmg "cp ${worksrcpath}/dmg/*/*.pkg ${worksrcpath}/pkg"
+        system -W ${worksrcpath}/dmg "hdiutil detach ${worksrcpath}/dmg/*"
+        system -W ${worksrcpath}/pkg "mv ${worksrcpath}/pkg/*.pkg ${worksrcpath}/pkg/jdk${ver}.pkg"
+        system -W ${worksrcpath}/pkg "${prefix}/bin/xar -xf ${worksrcpath}/pkg/jdk${ver}.pkg -C ${worksrcpath}/files"
+        system -W ${worksrcpath} "cpio -i < ${worksrcpath}/files/jdk180201.pkg/Payload"
+    }
+    worksrcdir          jdk${ver}
+} elseif {${configure.build_arch} eq "arm64"} {
+    set ver 8.54.0.21
+    set openjdk_ver
+    master_sites        https://cdn.azul.com/zulu/bin/
+    distname            zulu${ver}-ca-jdk${openjdk_ver}-macosx_aarch64
+    checksums           rmd160  cafab852e7263d3b1077b7b37f8d9019f4d3797a \
+                        sha256  2cf0af3118dd4b6244724308b190587804a76ad7919b6da5c377392a1d4f8817 \
+                        size    105926746
+
+    configure.cxx_stdlib libstdc++
+
+    worksrcdir          ${distname}/zulu-8.jdk
```

Comment:
So many differences in how these two archs are handled. Is there really no provider that offers both x86_64 and arm64 so that the two archs could be handled the same?

Reply:
Azul Zulu and Amazon Corretto do work on both but they don't support macOS X 10.7